### PR TITLE
HHH-10760 Add a getter to Index to expose the columnOrderMap

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/mapping/Index.java [hunk 1]
```diff
@@ -191,6 +191,10 @@ public int getColumnSpan() {
 		return columns.iterator();
 	}
 
+	public java.util.Map<Column, String> getColumnOrderMap() {
+		return columnOrderMap;
```
Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: what about return Collections.unmodifiableMap( columnOrderMap );?

Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: @dreab8 well, I thought about it but I decided to stay consistent with UniqueKey. I can change it if you prefer.

Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/mapping/Index.java]: I see anyway if I have to choose my vote is for unmodifiableMap :)
naturally if you have nothing against it
 Thanks

**Comment by gsmet** [comment]
@dreab8 done!

**Comment by dreab8** [comment]
Thanks @gsmet ,

rebased and pushed on master.

**Pull request closed without merging**